index.js

```diff
@@ -1163,18 +1195,16 @@
 
             var value = arguments[idx];
             if (checkTypes) {
-              var expType = expArgTypes[index];
-              if (!expType.test(value) ||
-                  isEmpty(satisfactoryTypes(name, constraints, expArgTypes,
-                                            expRetType, $typeVarMap,
-                                            expType, value, index))) {
+              if (!expTypes[index].test(value)) {
                 throw invalidValue(name,
                                    constraints,
-                                   expArgTypes.concat([expRetType]),
+                                   expTypes,
                                    value,
                                    index,
                                    determineActualTypesLoose([value]));
               }
+              satisfactoryTypes(name, constraints, expTypes,
+                                $typeVarMap, value, index);
```

Comment:
It turns out this function is only being called for side effects! At least now it's clear that this is the case.